controllers/dockercluster_controller.go

```diff
@@ -99,29 +98,28 @@ func (r *DockerClusterReconciler) Reconcile(req ctrl.Request) (_ ctrl.Result, re
 
 	// Handle deleted clusters
 	if !dockerCluster.DeletionTimestamp.IsZero() {
-		return reconcileDelete(dockerCluster, lbsvc)
+		return reconcileDelete(dockerCluster, externalLoadBalancer)
 	}
 
 	// Handle non-deleted clusters
-	return reconcileNormal(dockerCluster, lbsvc)
+	return reconcileNormal(dockerCluster, externalLoadBalancer)
 }
 
-func reconcileNormal(dockerCluster *infrav1.DockerCluster, lbsvc *docker.LoadBalancer) (reconcile.Result, error) {
+func reconcileNormal(dockerCluster *infrav1.DockerCluster, externalLoadBalancer *docker.LoadBalancer) (ctrl.Result, error) {
 	// If the DockerCluster doesn't have finalizer, add it.
 	if !util.Contains(dockerCluster.Finalizers, infrav1.ClusterFinalizer) {
 		dockerCluster.Finalizers = append(dockerCluster.Finalizers, infrav1.ClusterFinalizer)
 	}
 
 	//Create the docker container hosting the load balancer
-	n, err := lbsvc.Create()
-	if err != nil {
-		return reconcile.Result{}, errors.Wrapf(err, "failed to reconcile load balancer for DockerCluster %s/%s", dockerCluster.Namespace, dockerCluster.Name)
+	if err := externalLoadBalancer.Create(); err != nil {
```

Comment:
Is this always going to create? Should this have some check before hand?

Reply:
no, it is not necessary. 
The `externalLoadBalancer` is responsible for the docker container and `Create` already implements "Create if not exists"

Reply:
ah, yep, got it. Thanks!